metadata_fetcher/fetchers/oai_fetcher.py

```diff
@@ -102,15 +102,16 @@ def increment(self, http_resp):
 
         return
 
-
     def json(self):
-        if not self.oai.get('resumption_token'):
-            return None
-
-        return json.dumps({
+        current_state = {
             "harvest_type": self.harvest_type,
             "collection_id": self.collection_id,
             "write_page": self.write_page,
             "harvest_data": self.oai
-        })
+        }
+        if not self.oai.get('resumption_token'):
+            finished = {"finished": True}
+            finished.update(current_state)
+            return json.dumps(finished)
```

Comment:
This is a little confusing as is. How about more simply: 

```
        if not self.oai.get('resumption_token'):
            current_state.update({"finished": True})

        return json.dumps(current_state)
```

Reply:
oh. yeah. totally! thanks @barbarahui